llvm/lib/Target/AIE/AIE2InstructionSelector.cpp

```diff
@@ -3557,10 +3557,29 @@ LoadStoreOpcodes AIE2InstructionSelector::getLoadStoreOpcode(
                 /*OffsetOpcode=*/AIE2::VLDA_dmw_lda_am_ag_idx_imm};
       }
       if (RBID == AIE2::VRegBankID) {
-        FitsImmediateRange = checkImmediateRangeSplitting<11, 32, 32>(Offset);
-        return {/*ISelOpcode=*/AIE2::VLDA_dmw_lda_w_ag_idx_imm,
-                FitsImmediateRange,
-                /*OffsetOpcode=*/AIE2::VLDA_dmw_lda_w_ag_idx_imm};
+        unsigned OffsetOpcode;
+        // First try if the Instruction can be selected as multi-slot offset
+        // load
+        if (checkImmediateRangeSplitting<8, 32, 32>(Offset)) {
+          FitsImmediateRange = true;
+          ISelOpcode = OffsetOpcode = AIE2::VLD_idx_imm_3x32_pseudo;
+        } else if (checkImmediateRange<8, 32>(Offset)) {
+          // When Offset is positive and one of the offset is in range of SlotB
```

Comment:
Nit: I found the comments about positive/negative a bit confusing. I think we are just trying to find out which of `Offset` or `Offset+32` fits for VLDB. And we then assume the one that does not fit will fit for VLDA. Why do we need to check below for `(*Offset).isNegative()`?

Reply:
When we are loading 512 bit , we convert that in two 256 bit load i.e from `offset & offset+32`

When offset is negative we will come to that check only when Offset is **negative** and over the min range , but if number is negative then `Offset + 32` might be in range for Multi-Slot Pseudo. To be on safer side I added a check of `isNegative` and I felt the code was more reliable.